sentinelhub/download/sentinelhub_client.py

```diff
@@ -61,9 +74,12 @@ def _execute_download(self, request):
                 LOGGER.debug('%s: Sleeping for %0.2f', thread_name, sleep_time)
                 time.sleep(sleep_time)
 
-    def _execute_with_lock(self, thread_unsafe_function, *args, **kwargs):
+    def _execute_thread_safe(self, thread_unsafe_function, *args, **kwargs):
         """ Executes a function inside a thread lock and handles potential errors
         """
+        if self.lock is None:
```

Comment:
No, for a lock to be effective you have to create `1` lock and use it for all threads. If you create and use a different lock for each thread then such locks won't have any effect.

The idea is that a lock is created in `SentinelHubDownloadClient.download` method, which is the only one that has to download multiple things at once. Then here each thread can use the same `self.lock` object.

I see how this could be misinterpreted but if here `self.lock` is `None` this means that we didn't have to create any locks and therefore aren't using multiple threads. Hence we can just call the function directly.